modules/qc/bam_qc_snpsniffer.jst

```diff
@@ -0,0 +1,172 @@
+    - qc/snpSniffer/{{ project }}_SnpSniffer_AllPairs_Summary.tsv
+    - qc/snpSniffer/{{ project }}_SnpSniffer_PossibleMatchErrors_Summary.tsv
+    - qc/snpSniffer/{{ project }}_SnpSniffer_HetRate_Summary.tsv
+    - qc/snpSniffer/{{ project }}_SnpSniffer_PossibleCrossContamination_Summary.tsv
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByPatientPairType_All.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByAssayPairType_All.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByPatientPairType_SharedCalls100plus.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByAssayPairType_SharedCalls100plus.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByPatientPairType_SharedCalls50plus.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByPatientPairType_SharedCalls20plus.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByAssayPairType_SharedCalls50plus.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_MatchRatio_ByAssayPairType_SharedCalls20plus.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_HetRate_ByAssayType_All.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_Genotypes_ByAssayType_All.png
+    - qc/snpSniffer/{{ project }}_SnpSniffer_Mismatch_Summary.tsv
+  cpus: 1
+  walltime: "0:10:00"
+  cmd: |
+    set -eu
+    set -o pipefail
+
+    module load {{ constants.tools.snpSniffer.module }}
+
+    mkdir --parents temp/snpsniffer
+    mkdir --parents qc/snpSniffer
+
+    {# Create an empty version of the database #}
+    rsync {{ constants.phoenix.snpSniffer_database }} temp/snpsniffer/SnpSniffer_DB.ini
+
+    {# Collect all of the library level snpSniffer vcf results into the temp directory #}
+    {% for sample in samples.values() %}
+      {% for sample_lb, sample_rgs in sample.read_groups.values()|groupby('rglb') %}
+        rsync {{ sample.gltype }}/alignment/*/{{ sample.name }}/stats/{{ sample.name }}_{{ sample_lb }}.*.bam.snpSniffer.vcf temp/snpsniffer/
+      {% endfor %}
+    {% endfor %}
+
+    cd temp/snpsniffer
+
+    {# Add genotypes to the database #}
+    for line in `ls *snpSniffer.vcf`
+    do
+      java -jar $(which snpSniffer.jar) -add ${line} SnpSniffer_DB.ini
+    done
+
+    {#
+      When -expected or -NotExpect is queried a full comparison is produced to a
+      file called snpSniffer_output.txt, that is used for all summaries
+    #}
+    java -jar $(which snpSniffer.jar) -expected _ 2 SnpSniffer_DB.ini > Temp_Expected_Results.txt
+
+    {# Remove the unwanted file #}
+    rm Temp_Expected_Results.txt
+
+    {#
+      Preprocess the generic full comparison output
+      Replace unwanted features with tab-separtor
+      Create sorted list of line by line comparisons and drop duplicates
+    #}
+    sed 's/ & /\'$'\t''/g' snpSniffer_output.txt \
+      | \
+      sed 's/ count=/\'$'\t''/g' \
+      | \
+      sed 's/ match=/\'$'\t''/g' \
+      | \
+      sed 's/ ratio=/\'$'\t''/g' \
+      | \
+      sed 's/.bwa.bam.snpSniffer//g' \
```

Comment:
This will break for the bowtie alignments from ChipSeq, star-fusion, cellRanger, starSolo.

Passing the aligner variable might fix and then one of the lines can be removed

Reply:
also in the het section